src/thermodynamics.jl

```diff
@@ -0,0 +1,216 @@
+"""
+    get_thermodynamics!(
+        column::ColumnVariables{NF},
+        model::PrimitiveEquationModel{NF},
+    )
+"""
+function get_thermodynamics!(
+    column::ColumnVariables{NF},
+    model::PrimitiveEquationModel{NF},
+) where {NF<:AbstractFloat}
+    # Calculate thermodynamic quantities at full levels
+    saturation_vapour_pressure!(column, model)
+    saturation_specific_humidity!(column, model)
+    dry_static_energy!(column, model)
+    moist_static_energy!(column, model)
+    saturation_moist_static_energy!(column, model)
+
+    # Interpolate certain variables to half-levels
+    interpolate!(column, model)
+
+    return nothing
+end
+
+"""
+    interpolate!(
+        column::ColumnVariables{NF},
+        model::PrimitiveEquationModel,
+    )
+"""
+function interpolate!(
+    column::ColumnVariables{NF},
+    model::PrimitiveEquationModel,
+) where {NF<:AbstractFloat}
+    @unpack humid, humid_half = column
+    @unpack sat_humid, sat_humid_half = column
+    @unpack dry_static_energy, dry_static_energy_half = column
+    @unpack sat_moist_static_energy, sat_moist_static_energy_half = column
+
+    for (full, half) in (
+        (humid, humid_half),
+        (sat_humid, sat_humid_half),
+        (dry_static_energy, dry_static_energy_half),
+        (sat_moist_static_energy, sat_moist_static_energy_half),
+    )
+        interpolate!(full, half, column, model)
+    end
+
+    return nothing
+end
+
+"""
+    interpolate!(
+        A_full_level::Vector{NF},
+        A_half_level::Vector{NF},
+        column::ColumnVariables{NF},
+        model::PrimitiveEquationModel,
+    )
+
+Given some generic column variable A defined at full levels, do a linear interpolation in
+log(σ) to calculate its values at half-levels.
+"""
+function interpolate!(
+    A_full_level::Vector{NF},
+    A_half_level::Vector{NF},
+    column::ColumnVariables{NF},
+    model::PrimitiveEquationModel,
+) where {NF<:AbstractFloat}
+    @unpack nlev = column
+    @unpack σ_levels_full, σ_levels_half = model.geometry
+
+    # For A at each full level k, compute A at the half-level below, i.e. at the boundary
+    # between the full levels k and k+1.
+    for k = 1:nlev-1
+        A_half_level[k] =
+            A_full_level[k] +
+            (A_full_level[k+1] - A_full_level[k]) *
+            (log(σ_levels_half[k+1]) - log(σ_levels_full[k])) /
+            (log(σ_levels_full[k+1]) - log(σ_levels_full[k]))
+    end
+
+    # Compute the values at the surface separately
+    A_half_level[nlev] =
+        A_full_level[nlev] +
+        (A_full_level[nlev] - A_full_level[nlev-1]) *
+        (log(0.99) - log(σ_levels_full[nlev])) /
```

Comment:
`log0.99 = convert(NF,log(0.99))` directly after the `@unpack`s instead of `log(0.99)` here?

Reply:
How about just doing `log(NF(0.99))` here? It's outside of the loop so only gets executed once.

Reply:
Yes, also fine! Any clue why 0.99? Seems like a weird hard-coded constant to me?!

Reply:
Yeah, good question, I can't immediately see me why you wouldn't just use `log(σ_levels_half[nlev])`, i.e. `0`.

Reply:
Probably some weird trick that no one bothered commenting...:see_no_evil: